cs/ccxt/base/Throttler.cs

```diff
@@ -18,30 +20,40 @@ public Throttler(dict config)
             {"refillRate",1.0},
             {"delay", 0.001},
             {"cost", 1.0},
-            {"tokens", 0},
-            {"maxCapacity", 2000},
+            {"tokens", 0.0},
+            {"maxLimiterRequests", 2000},
             {"capacity", 1.0},
+            {"algorithm", "leakyBucket"},
+            {"rateLimit", 0.0},
+            {"windowSize", 60000.0},
+            // maxWeight should be set in config parameter for rolling window algorithm
         };
         this.config = extend(this.config, config);
 
     }
 
-    private async Task loop()
+    private async Task leakyBucketLoop()
     {
         var lastTimestamp = milliseconds();
         while (this.running)
         {
             // do we need this check here?
-            if (this.queue.Count == 0)
+            lock (queueLock)
+            {
+                if (this.queue.Count == 0)
+                {
+                    this.running = false;
+                    continue;
+                }
+            }
```

Comment:
@carlosmiei Unless I miss smth, do we need this check here? seems not needed, as I can't think of scenario where it can start the `while` loop where queue can be 0. the removal of item (from queue) happens down below body and from there it's breaks loop (if count is 0), but why it's needed in the top?

```suggestion
            lock (queueLock)
            {
                if (this.queue.Count == 0)
                {
                    this.running = false;
                    continue;
                }
            }
```